Content.Server/Lock/LockComponent.cs

```diff
@@ -20,6 +20,7 @@ public class LockComponent : Component
         public override string Name => "Lock";
 
         [ViewVariables(VVAccess.ReadWrite)] [DataField("locked")] public bool Locked { get; set; } = true;
+        [ViewVariables(VVAccess.ReadWrite)] [DataField("lockOnClick")] public bool LockOnClick { get; set; } = false;
```

Comment:
LockComp/System itself should not be handling this, other systems should be toggling/listening for lock instead. make a more generalized component & system called something like LockOnClick, see my comment above on mgs' review

Reply:
Could you please provide more information about what LockOnClick should do? 

- Should I delete all interactions logic from LockSystem and recreate it in LockOnClickSystem? 
- Should I create new event that is raised by LockOnClick? So it will work like: 
  `LockOnClickSystem => LockOnClickEvent => LockSystem => LockToggleEvent`
- What fields should LockOnClick have?
- Does LockOnClick should support lockers? Or I need create LockOnClickLockers to handle open-closed state of the locker?

Reply:
not sure if you saw in discord;
![image](https://user-images.githubusercontent.com/19853115/129630503-26c4e664-3021-40b9-a329-9560bdfb9a75.png)